mimic/rest/auth_api.py

```
@@ -48,75 +112,23 @@ def get_token_and_service_catalog(self, request):
-                         session.tenant_id,
-                         prefix_map, base_uri_from_request(request))),
-                    prefix_for_endpoint=lookup,
-                    response_token=session.token,
-                    response_user_id=session.user_id,
-                    response_user_name=session.username,
-                )
-                return json.dumps(result)
-
-        username_generator = (
-            lambda exception: "Tenant with Name/Id: '{0}' is not valid for "
-                              "User '{1}' (id: '{2}')".format(
-                                  exception.desired_tenant,
-                                  exception.session.username,
-                                  exception.session.user_id))
-
-        cred_mappings = {
-            'passwordCredentials':
-                (PasswordCredentials, username_generator),
-            'RAX-KSKEY:apiKeyCredentials':
-                (APIKeyCredentials, username_generator),
-            'token':
-                (TokenCredentials,
-                 lambda e: (
-                     "Token doesn't belong to Tenant with Id/Name: "
-                     "'{0}'".format(e.desired_tenant)))
-        }
-
-        relevant_cred_keys = set(cred_mappings).intersection(
-            set(content['auth']))
-
-        if relevant_cred_keys:
-            cred_type, error_handler = cred_mappings[relevant_cred_keys.pop()]
-            try:
-                cred = cred_type.from_json(content)
-            except Exception:
-                pass
-            else:
-                return format_response(cred, error_handler)
+            pass
+        else:
+            for cred_type in (PasswordCredentials, APIKeyCredentials,
+                              TokenCredentials):
+                if cred_type.type_key in content['auth']:
+                    try:
+                        cred = cred_type.from_json(content)
+                    except Exception:
+                        pass
```

`except:pass` is always a real red flag :).  `from_json` should raise a *specific* exception type, and this `except` should catch only that type.